guides/source/routing.md

```diff
@@ -618,14 +653,16 @@ This will enable Rails to recognize paths such as `/comments/new/preview` with G
 
 TIP: If you find yourself adding many extra actions to a resourceful route, it's time to stop and ask yourself whether you're disguising the presence of another resource.
 
+It is possible to customize the default routes and helpers generated by `resources`, see [customizing resourceful routes section](#customizing-resourceful-routes) for more.
+
 Non-Resourceful Routes
 ----------------------
```

Comment:
This section has a lot of subsection, some which are pretty short compared to other guides.
Could we maybe group some of these and use a smaller heading for the grouped sections?
For example group the segments (I've added more sections here) and constraints sections?
Maybe something like:
- Routing Segments
  - Bound Parameters
  - Dynamic Segments
  - Static Segments
  - Wildcard Segments
  - Format Segments
  - The Query String
  - Defining Default Parameters
  - Naming Routes
  - Unicode Character Routes
- Routing Contraints
  - HTTP Verb Constraints
  - Segment Constraints
  - Request-Based Constraints
  - Advanced Constraints
- Redirection
- Using root
- Routing to Rack Applications
- Direct Routes
- Using resolve

Reply:
Hi @p8, this section does have a lot of sub-sections and we could group them. I think the next heading level h4 won't show up in the nice fancy scrolling index thing on the right? Surfacing these "features" there feels helpful to the readers in discovering the content. Otherwise they'd have to just scroll around to find it. My sense is "Segments" and "Constraints" headings aren't revealing enough unless you already know a lot about routing.  What do you think? I'm open to grouping it like you showed above. But thought I'd mention the index thing.

Reply:
I don't have a strong opinion on this. Let's keep it as is.